mp/importer/tests/test_batcher.py

```diff
@@ -0,0 +1,124 @@
+import os
+from unittest import TestCase
+import collections
+from collections import namedtuple
+from collections import Iterable
+
+def make_test_context_1():									# creates a "provisional" context 
+    context = namedtuple('Context', """
+        start_batch
+        batch_size """)
+    context.batch_size = 2									# batcher will read it from context in the script
+    context.start_batch = 0
+    return context
+
+def make_test_context_2():									 
+    context = namedtuple('Context', """
+        start_batch
+        batch_size """)
+    context.batch_size = 2									
+    context.start_batch = 2
+    return context
+
+# test function (it will be the import function) - generalize for different types of iterable - now writes results in a list with nested lists (to be changed if needed)
+def process_one(item):
+    result = []
+    if isinstance(item, str):
+        result.append(item.upper())
+    else:
+        for x in item:
+            result.append(x.upper()) 
+    return result
+
+def process_one_simple(item):
+    result = []
+    result.append(item.upper())
+    return result
+# remember that batch_item is a piece of an iterable, done like this batch_item = iterable[n*batch_size:batch_size+n*batch_size] - can be done only for objects that have __getitem__
+
+class TestBatcher(TestCase):
+    # 1- test that the function makes batches of a given size
+    def test_Make_Batches(self):
+        input_string = 'abcdefg'									
+        context = make_test_context_1()
+        from .. import batcher									
+        result = batcher.run_in_batches(context, input_string, process_one_simple)				
+        self.assertEqual(result,[['AB'], ['CD'], ['EF'], ['G']])
+    #
+    # 2- test that the function starts being executed from a specified batch number (batches of given size)
+    def test_Start_from_Batch(self):
+        input_string = 'abcdefg'									
+        context = make_test_context_2()
+        from .. import batcher									
+        result = batcher.run_in_batches(context, input_string, process_one_simple)				
+        self.assertEqual(result,[['EF'], ['G']])
+    #
+    # 3- test if inputs of the function are iterables or not
+    def test_Batcher_iterables(self):
```

Comment:
I am not convinced that it is useful to test different kinds of iterables. The iterable interface in python is very well defined and any code that works with that interface should work with any iterable.

/me reads the python documentation ...  ... 

Though, I realize that I was definitely not "precise" enough;) what we are aiming for is code which works with "iterator" objects (https://docs.python.org/2/glossary.html#term-iterator). The only thing an iterator object has is a next() method.

```
>>> mylist = [1,2,3]
>>> myiter = iter(mylist)
>>> myiter.next()
1
```